tests/README.md

```diff
@@ -0,0 +1,128 @@
+set(TEST_NAME ble-service-your-service-unittest)
+
+add_executable(${TEST_NAME})
+
+target_include_directories(${TEST_NAME}
+    PRIVATE
+        .
+        # change the path to where your includes are
+        ${SERVICES_PATH}/YourService/include
+)
+
+target_sources(${TEST_NAME}
+    PRIVATE
+        # replace with the source file you have your unit test in
+        test_YourService.cpp
+        # replace the source file with the one that compiles your service
+        ${SERVICES_PATH}/YourService/source/YourService.cpp
+)
+
+target_link_libraries(${TEST_NAME}
+    PRIVATE
+        # add any stubs, headers or mocks you need here
+        # you will most likely want to include:
+        # mbed-os-fakes-ble
+        gmock_main
+)
+
+add_test(NAME "${TEST_NAME}" COMMAND ${TEST_NAME})
+```
+
+Inside CMake files for unit tests there are two special variables you can use that are set by the top level
+unit test CMakeLists.txt: `MBED_PATH`, `SERVICES_PATH`. These point to mbed-os and the services directory.
+
+## Stubs, mocks and fakes in mbed-os
+
+To make it easier to test your service mbed-os provides some ready made CMake libraries you can add in your
+`target_link_libraries` section (see template above). Libraries can be object files that contains stubs, mocks
+and fakes but also sets of include paths - this way you can gain access to normal headers from mbed-os.
+
+All the libraries available are in the `CMakeLists.txt` files in `mbed-os/UNITTESTS/stubs` and
+`mbed-os/UNITTESTS/fakes`.
+
+The most common ones you might need are `mbed-os-fakes-ble` and `mbed-os-fakes-event-queue`.
+
+### mbed-os-fakes-ble
+
+This library provides a fake BLE implementation that uses mocks instead of real BLE components for `Gap`,
+`GattServer`, `GattClient`, `SecurityManager`.
+
+There is no need to initialise the instance, it is ready to go and can get used just like normal BLE instance:
+```
+BLE ble = &BLE::Instance();
+```
+
+You can retrieve all the BLE APIs from the instance just like with a normal one:
+
+```
+Gap &gap = ble->gap();
+GattClient &client = ble->gattClient();
+GattServer &server = ble->gattServer();
+SecurityManager &sm = ble->securityManager();
+```
+
+But these are instead google mocks. This means you can set expectations on them. 
+
+```
+EXPECT_CALL(ble::gap_mock(), reset());
+```
+
+This will set up an expectations that at some point during the test the Gap::reset method will be called.
+
+The way google test works means that if you set any expectations on your mocks they must be destroyed at
+the end of each test. This is done through the fake BLE instance special method:
+
+```
+ble::reset_mocks();
+```
+
+### mbed-os-fakes-event-queue
+
+This is a fake event queue that doesn't bring in any dependencies from mbed-os. It's API is simplified
+and it only offers limited functionality.
+
+If you choose to use it you must not also include a library that brings in real headers for the event
+queue as they would conflict.
```

Comment:
Is this true. Did you not include `events/EventQueue` in your illustrative example?

Reply:
it is true. that header is a fake header

Reply:
What's the full path of the fake header in your [cmake_lists](https://github.com/paul-szczepanek-arm/mbed-os/tree/cmake-tests) branch in mbed-os?

Reply:
I don't understand the question, look inside UNITTESTS/fakes/ that's where the library is, it has it's own header

Reply:
I think this line should be clarified. As a user, I would like to know if I can include the event header in my service with a `events/EventQueue.h` and which API is not available in the fake. 

Also what would be the issue if another library includes `events/EventQueue.h` ? It is not possible to propagate the fake event include path when its builded ? That would be a minor issue if it was the case as we try to avoid real dependency in unit tests.